Assets/MixedRealityToolkit.SDK/Features/Utilities/InputUtils.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.WindowsDevicePortal;
+using System.Runtime.InteropServices.WindowsRuntime;
+using UnityEngine;
+using UnityEngine.Experimental.XR;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Utilities for accessing input state.
+    /// </summary>
+    public static class InputUtils
+    {
+        /// <summary>
+        /// Gets the ray representing the position and direction of the user's head.
+        /// </summary>
+        /// <returns>The ray the head gaze</returns>
+        public static Ray GetHeadGazeRay()
```

Comment:
What about if I already have a pointer and I want to get it's ray?
We're looping through potentially all input sources every function call

Reply:
To get the ray for a pointer you can just grab the position and rotation of a pointer. I do think that we could cache the right and left hands, but not convinced it's such a massive perf hit. Will check.

Reply:
Okay, ran a profiler. The scene I have is slow, but this is because of UGUI.Rendering.UpdateBatches, most likely because I'm changing the text of a canvas every frame. Also I'm allocating some stuff every frame, probably the StringBuilder. So, nothing aggregious in accessing the controllers.

Reply:
Hm now that I've played more with the scene, I'm noticing that the framerate on the example scene does drop to 40. I'll file a bug on that, don't really want to ship with that. Worst case we will remove the code that renders the status of controllers.

Reply:
#5996

Reply:
That's great you tested the example scene!

I was actually referring to people consuming the API. Sort of like CoreServices or other utils, this could be used by many different features in an app and thus the loop is applied by every call. I meant the only workaround is if they had a reference to the controller/pointer already